src/resultscallercalleepage.cpp

```diff
@@ -234,21 +233,17 @@ void ResultsCallerCalleePage::onSourceMapContextMenu(QPoint point)
 
     QMenu contextMenu;
     auto* viewCallerCallee = contextMenu.addAction(tr("Open in Editor"));
-    auto* action = contextMenu.exec(QCursor::pos());
-    if (action == viewCallerCallee) {
-        emit navigateToCode(location.path, location.lineNumber, 0);
-    }
-}
+    connect(viewCallerCallee, &QAction::triggered, this,
+            [this, location] { emit navigateToCode(location.path, location.lineNumber, 0); });
+
+    auto line = index.data(SourceMapModel::FileLineRole).value<Data::FileLine>();
+    auto disassemblyAction = contextMenu.addAction(tr("Disassembly"));
+    connect(disassemblyAction, &QAction::triggered, this, [this, line] {
+        if (m_symbol.isValid())
```

Comment:
use the validity to enable/disable the action in the first place - to prevent the slot from being called in situations where your code would silently not do anything

Reply:
also, it's imo not nice to have an action called "disassembly" silently not do anything if the disassembly isn't showing anything relating to the symbol at hand. that would happen with this code, or? since it only scrolls - it doesn't reload the data in the disassembly view?

Reply:
I think I (hopefully?) misunderstand the code. can you clarify: if I select this on a different entry than is currently shown in the disassembly view, it will select that one first and then scroll to the line afterwards?

Reply:
ugh, there is a lot of bad naming going on `ResultsCallerCalleePage::jumpToSourceCode` calls `ResultsPage::onJumpToSourceCode` which calls `ResultsDisassemblyPage::onJumpToDisassembly` which will actually change the file. Afterwords `ResultsPage::onJumpToSourceCode` calls `ResultsDisassemblyPage::jumpToSourceLine`

Reply:
so yeah it will change the file and then scroll to the line

Reply:
would be nice if you could clean this up eventually